### ISSUE TITLE
VB -> C#: for loop with Int16 index needs explicit cast in generated code declaring loopTo variable

### ISSUE THREAD

**Issue by jamesmanning** [icsharpcode/CodeConverter#607]
### Input code
```vbnet
    Sub DummyMethod()
        Dim someArray = New Integer() { 1, 2, 3}
        For index As Int16 = 0 To someArray.Length - 1
            Console.WriteLine(index)
        Next
    End Sub
```

### Erroneous output
```C#
        public void DummyMethod()
        {
            var someArray = new int[] { 1, 2, 3 };
            for (short index = 0, loopTo = someArray.Length - 1; index <= Conversions.ToShort(loopTo); index++)
                Console.WriteLine(index);
        }
```

### Expected output

Assuming loopTo is correctly being declared a short, we can/should drop the Conversions.ToShort in the check clause and move it to the assignment clause AFAICT? Removing it from the check clause is optional, the compiler error is due to trying to implicitly assign the 32-bit integer of someArray.Length - 1 to the 16-bit short loopTo

```C#
        public void DummyMethod()
        {
            var someArray = new int[] { 1, 2, 3 };
            for (short index = 0, loopTo = Conversions.ToShort(someArray.Length - 1); index <= loopTo; index++)
                Console.WriteLine(index);
        }
```

### Details
* VS 2019 Community with Code Converter extension 8.1.6.0

**Comment by GrahamTheCoder** [comment]
This was fixed to match your expected output for https://github.com/icsharpcode/CodeConverter/issues/602, sorry it didn't get released in time for your conversion. I'll try to get it released this weekend.

I hadn't carefully checked this case, but I've now checked the decompilation eagerly coerces the loopTo variable to the control variable type.

There's one remaining subtle difference from the decompilation, which is that it appears VB uses unchecked addition. I'm going to add this as a note in known limitations #16
